### PR TITLE
NAS-132131 / 25.04 / Add docker nvidia status

### PR DESCRIPTION
**Pull request by denysbutenko**
**Changes:**

Add support for `docker.nvidia_status` endpoint

**Testing:**

Check the Apps page and settings.

### CODE REVIEW

File: src/app/pages/apps/store/docker.store.ts [hunk 1]
```diff
@@ -38,7 +39,8 @@ export class DockerStore extends ComponentStore<DockerConfigState> {
   readonly dockerConfig$ = this.select((state) => state.dockerConfig);
   readonly selectedPool$ = this.select((state) => state.dockerConfig?.pool || null);
   readonly nvidiaDriversInstalled$ = this.select((state) => state.nvidiaDriversInstalled);
-  readonly lacksNvidiaDrivers$ = this.select((state) => state.lacksNvidiaDrivers);
+  readonly lacksNvidiaDrivers$ = this.select((state) => state.nvidiaStatus !== DockerNvidiaStatus.Absent);
```
Comment (undsoft) [inline src/app/pages/apps/store/docker.store.ts:42]: I don't think name matches the condition.

**Review by undsoft — COMMENTED** [review]
Middleware behavior doesn't match my expectations.
On a CI machine, the initial status is `NOT_INSTALLED`, when I install drivers it becomes `ABSENT`.
Shouldn't it be the other way around, @themylogin?

**Comment by themylogin** [comment]
@undsoft you are correct, fixed https://github.com/truenas/middleware/pull/14872

**Comment by denysbutenko** [comment]
@undsoft now it returns `ABSENT` on nightlies and this checkbox never appears. 

For testing:
```
midclt call test.remove_mock docker.nvidia_status '[]'
midclt call test.set_mock docker.nvidia_status '[]' '{"return_value": {"status":"NOT_INSTALLED"}}'
midclt call docker.nvidia_status
```

**Review by undsoft — APPROVED** [review]
Looks good.

@themylogin Please note that it's possible to have a mismatch beween `nvidia` property in `docker.config` and `docker.nvidia_status`. If you force driver installation on a system that doesn't have nvidia cards, the former will return `true`, while the latter returns `ABSENT`.
This is fine (good actually) for purposes of UI behavior, but if you fix it in the future, UI may behave unexpectedly.

**Comment by denysbutenko** [comment]
backport

**Pull request merged**